src/domain/session/room/timeline/MessageBodyBuilder.js

```diff
@@ -0,0 +1,161 @@
+import { linkify } from "./linkify.js";
+
+export class MessageBodyBuilder {
+
+    constructor(message = []) {
+        this._root = message;
+    }
+
+    fromText(text) {
+        const components = text.split("\n");
+        components.flatMap(e => ["\n", e]).slice(1).forEach(e => {
+            if (e === "\n")
```

Comment:
please use brackets for all block statement (this should really be in a style guide, sorry)